backend/src/main/java/asegroup1/api/models/LandRegistryQueryConstraint.java

```diff
@@ -0,0 +1,175 @@
+package asegroup1.api.models;
+
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+/**
+ * Contains the constraints to be passed
+ * 
+ * @author Rikkey Paal
+ */
+public class LandRegistryQueryConstraint {
+
+	private HashMap<String, Constraint> constraints;
+
+	public LandRegistryQueryConstraint() {
+		constraints = new HashMap<>();
+	}
+
+	private void addAddrConstraint(String name, String value) {
+		constraints.put(name, new AddrConstraint(name, "\"" + value + "\""));
+	}
+
+	public void setPrimaryHouseName(String primaryHouseName) {
+		addAddrConstraint("paon", primaryHouseName);
+	}
+
+	public void removePrimaryHouseName(String primaryHouseName) {
+		constraints.remove("paon");
+	}
+
+	public void setSecondaryHouseName(String secondaryHouseName) {
+		addAddrConstraint("saon", secondaryHouseName);
+	}
+
+	public void setStreetName(String streetName) {
+		addAddrConstraint("street", streetName);
+	}
+
+	public void setTownName(String townName) {
+		addAddrConstraint("town", townName);
+	}
+
+	public void setDistrict(String district) {
+		addAddrConstraint("district", district);
+	}
+
+	public void setCounty(String county) {
+		addAddrConstraint("county", county);
+	}
+
+	public void setPostCode(String postCode) throws InvalidParameterException {
+		if (postCode.charAt(postCode.length() - 4) != 32) {
+			throw new InvalidParameterException("Post Code: must contain a space in the correct position");
+		} else {
+			addAddrConstraint("postcode", postCode.toUpperCase());
+		}
+	}
+
+
+	private void addTransConstraint(String name, String value) {
+		constraints.put(name, new TransConstraint(name, value));
+	}
+
+	public void setPropertyType(PropertyType propertyType) {
+		addTransConstraint("propertyType", parseEnumValue("lrcommon", propertyType.toString()));
+	}
+
+	public void setNewBuild(Boolean newBuild) {
+		addTransConstraint("newBuild", newBuild.toString());
+	}
+
+	public void setEstateType(EstateType estateType) {
+		addTransConstraint("estateType", parseEnumValue("lrcommon", estateType.toString()));
+	}
+
+	public void setTransactionCategory(TransactionCategory transactionCategory) {
+		addTransConstraint("transactionCategory", parseEnumValue("lrppi", transactionCategory.toString()));
+	}
+
+	public String buildQueryWhere() {
+		return "WHERE { \n" + buildQuerySelection() + "\n" + buildQueryColumns() + "\n}";
+	}
+
+	private String buildQuerySelection() {
+		
+		ArrayList<Constraint> constraintList = new ArrayList<>(constraints.values());
+		// order by type
+		Collections.sort(constraintList);
+		
+		StringBuilder whereStringBuilder = new StringBuilder();
+		for(Constraint constraint : constraintList) {
+			whereStringBuilder.append(constraint.toString() + " \n");
```

Comment:
This line should be `whereStringBuilder.append(constraint.toString() ).append(" \n");`